bedrock/bedrock_test.go

```diff
@@ -0,0 +1,284 @@
+		toolSpec, ok := converted[0].(*types.ToolMemberToolSpec)
+		require.True(t, ok)
+		assert.Equal(t, "get_weather", aws.ToString(toolSpec.Value.Name))
+		assert.Equal(t, "Get the current weather", aws.ToString(toolSpec.Value.Description))
+		require.NotNil(t, toolSpec.Value.InputSchema)
+	})
+
+	t.Run("convert multiple tools", func(t *testing.T) {
+		schema := &jsonschema.Schema{
+			Type:       "object",
+			Properties: map[string]*jsonschema.Schema{},
+		}
+
+		tools := []llm.Tool{
+			{
+				Name:        "tool1",
+				Description: "First tool",
+				Schema:      schema,
+			},
+			{
+				Name:        "tool2",
+				Description: "Second tool",
+				Schema:      schema,
+			},
+		}
+
+		converted := convertTools(tools)
+
+		require.Len(t, converted, 2)
+
+		toolSpec1, ok := converted[0].(*types.ToolMemberToolSpec)
+		require.True(t, ok)
+		assert.Equal(t, "tool1", aws.ToString(toolSpec1.Value.Name))
+
+		toolSpec2, ok := converted[1].(*types.ToolMemberToolSpec)
+		require.True(t, ok)
+		assert.Equal(t, "tool2", aws.ToString(toolSpec2.Value.Name))
+	})
+
+	t.Run("empty tools array", func(t *testing.T) {
+		tools := []llm.Tool{}
+		converted := convertTools(tools)
+		assert.Len(t, converted, 0)
+	})
+}
+
+func TestGetDefaultConfig(t *testing.T) {
+	t.Run("with custom output token limit", func(t *testing.T) {
+		b := &Bedrock{
+			defaultModel:     "anthropic.claude-3-5-sonnet-20241022-v2:0",
+			outputTokenLimit: 4096,
+		}
+
+		config := b.GetDefaultConfig()
+
+		assert.Equal(t, "anthropic.claude-3-5-sonnet-20241022-v2:0", config.Model)
+		assert.Equal(t, 4096, config.MaxGeneratedTokens)
+	})
+
+	t.Run("with default output token limit", func(t *testing.T) {
+		b := &Bedrock{
+			defaultModel:     "anthropic.claude-3-5-sonnet-20241022-v2:0",
+			outputTokenLimit: 0,
+		}
+
+		config := b.GetDefaultConfig()
+
+		assert.Equal(t, "anthropic.claude-3-5-sonnet-20241022-v2:0", config.Model)
+		assert.Equal(t, DefaultMaxTokens, config.MaxGeneratedTokens)
+	})
+}
+
+func TestInputTokenLimit(t *testing.T) {
+	tests := []struct {
+		name            string
+		defaultModel    string
+		inputTokenLimit int
+		expectedLimit   int
+	}{
+		{"custom limit", "anthropic.claude-3-5-sonnet-20241022-v2:0", 150000, 150000},
+		{"claude 3.5", "anthropic.claude-3-5-sonnet-20241022-v2:0", 0, 200000},
+		{"claude 3", "anthropic.claude-3-opus-20240229-v1:0", 0, 200000},
+		{"titan express", "amazon.titan-text-express-v1", 0, 8000},
+		{"titan lite", "amazon.titan-text-lite-v1", 0, 4000},
+		{"unknown model", "unknown.model", 0, 100000},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			b := &Bedrock{
+				defaultModel:    tt.defaultModel,
+				inputTokenLimit: tt.inputTokenLimit,
+			}
+
+			limit := b.InputTokenLimit()
+			assert.Equal(t, tt.expectedLimit, limit)
+		})
+	}
+}
+
+func TestCountTokens(t *testing.T) {
+	b := &Bedrock{}
+
+	tests := []struct {
+		name     string
+		text     string
+		minCount int
```

Comment:
Why a range? If we are going to test it we might as well get the right answer :P